inc/io.php

```diff
@@ -271,41 +305,45 @@ function io_deleteFromFile($file,$badline,$regex=false){
     // load into array
     if(substr($file,-3) == '.gz'){
         $lines = gzfile($file);
+    }else if(substr($file,-4) == '.bz2'){
+        $lines = bzfile($file, true);
     }else{
         $lines = file($file);
     }
 
     // remove all matching lines
     if ($regex) {
-        $lines = preg_grep($badline,$lines,PREG_GREP_INVERT);
+        if($maxlines > 0) {
```

Comment:
preg_replace supports a limit parameter, can't this '> 0' case be handled by the preg_replace in line 325